sources/packages/backend/libs/integrations/src/institution-integration/ier12-integration/ier12-file-detail.ts

```diff
@@ -341,6 +319,9 @@ export class IER12FileDetail implements IER12FileLine {
     record.append(this.getAwardDetails(FullTimeAwardTypes.BCAG), 14);
     record.append(this.getAwardDetails(FullTimeAwardTypes.SBSD), 14);
     record.append(this.getAwardDetails(FullTimeAwardTypes.BGPD), 14);
+    // Filler to populate the award details on 10th position.
+    record.appendStringWithFiller("", 4);
```

Comment:
trying to understand the difference between `appendOptionalStringWithFiller` and `appendStringWithFiller`, isn't the output for both the same?

Reply:
`appendOptionalStringWithFiller` is intended for optional fields which can be null at runtime.(it has an internal null check).
appendOptionalStringWithFiller uses appendStringWithFiller after performing null check.

![image](https://github.com/bcgov/SIMS/assets/54600590/613afbb2-7a89-420c-80bf-fc836b557d4e)